src/transforms/remap.rs

```diff
@@ -152,13 +156,25 @@ impl FunctionTransform for Remap {
                     output.push(event)
                 }
             }
-            Err(Terminate::Abort(_)) => {
+            Err(Terminate::Abort(error)) => {
                 emit!(&RemapMappingAbort {
                     event_dropped: self.drop_on_abort,
                 });
 
                 if !self.drop_on_abort {
                     output.push(original_event.expect("event will be set"))
+                } else {
+                    let mut erred_event = original_event.expect("event will be set");
+                    if let Event::Log(ref mut log) = &mut erred_event {
+                        log.insert(
+                            log_schema().metadata_key(),
+                            serde_json::json!({
+                                "error": error.to_string(),
+                                "component": self.component_key,
+                            }),
+                        );
+                    }
```

Comment:
I probably missed this earlier, but is there a reason we are only doing this for logs and not also metrics which `remap` can be used for?

Reply:
This type of data (e.g. error messages, etc) didn't really strike me as making sense to shove into metrics tags at the time, but I don't really have a strong opinion on that. If we think they'd be useful I'm happy to add some equivalent to metrics as well.

Reply:
Ah, yeah, that is a wrinkle if we allow routing of failed metrics. 🤔  Overall routing of failed metrics is probably of a less common use-case. I could see supporting it just for parity though, adding the metadata as metrics tags.